ts/src/gate.ts

```diff
@@ -5747,12 +5791,18 @@ export default class gate extends Exchange {
          * @param {object} params extra parameters specific to the gate api endpoint
          * @returns {object} a [ledger structure]{@link https://docs.ccxt.com/#/?id=ledger-structure}
          */
+        const response = await this.fetchRawRecords ('fetchLedger', code, since, limit, params);
+        const currency = this.currency (code);
+        return this.parseLedger (response, currency, since, limit);
+    }
+
```

Comment:
@ttodua there is no problem to abstract the common logic in a separate method when needed  but in this case you sure that the same set of endpoints fit both unified methods? (transfers and ledger) For instance after placing an order this is returned by fetchTransfers
```
gate.fetchTransfers ()
2023-06-23T09:29:29.608Z iteration 0 passed in 428 ms

         id |     timestamp |                 datetime | currency |    amount | fromAccount | toAccount | status
----------------------------------------------------------------------------------------------------------------
38660095041 | 1687512525757 | 2023-06-23T09:28:45.757Z |      LTC | 0.0001152 |             |           |       
38660095040 | 1687512525757 | 2023-06-23T09:28:45.757Z |      LTC |    0.0576 |             |           |       
38660095039 | 1687512525757 | 2023-06-23T09:28:45.757Z |     USDT |  4.994496 |             |           |       
3 objects
```
and it does not seem to be a transfer